### PR TITLE
AssetIndex: New PostgreSQL extension, closes #444

### LINKED ISSUE #444: Extension - AssetIndex - Integrate the PostgreSQL library
Issue #438 provided a library that supports PostgreSQL persistence. Integrate this library in a new `AssetIndex` extension.

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

### PR DESCRIPTION
**Pull request by DominikPinsel**
Based on PR #439

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

### CODE REVIEW

**Comment by DominikPinsel** [comment]
@ndr-brt  @alexandrudanciu @juliapampus This PR is based on PR #438  from Denis. Maybe you can take a look at this PR, too. It uses the library from PR #438 and implements an `AssetIndex` with it. 

I moved most of the Asset PostgreSQL code into a second library to have it reusable in other extensions. For example by reusing it to implement the `AssetLoader` interface, which I do not see in the same `AssetIndex` extension. 

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/connection/ConnectionFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        Objects.requireNonNull(connectionFactoryConfig, "connectionFactoryConfig");
+
+        this.jdbcUrl = Objects.requireNonNull(connectionFactoryConfig.getUri()).toString();
+        this.properties = getProperties(connectionFactoryConfig);
+        this.autoCommit = connectionFactoryConfig.getAutoCommit();
+    }
+
+    private Properties getProperties(ConnectionFactoryConfig connectionFactoryConfig) {
+        Properties properties = new Properties();
+        addProperty(properties, "user", connectionFactoryConfig.getUserName());
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/connection/ConnectionFactoryImpl.java:43]: Are these random string values or is there a rule/expectation from another module? Are the values maybe specified by another library (java.sql)?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BigDecimalArgumentHandler.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum BigDecimalArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BigDecimalArgumentHandler.java:22]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BooleanArgumentHandler.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum BooleanArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BooleanArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ByteArgumentHandler.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum ByteArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ByteArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BytesArgumentHandler.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum BytesArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BytesArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DateArgumentHandler.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+
+public enum DateArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DateArgumentHandler.java:23]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DoubleArgumentHandler.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum DoubleArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DoubleArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/FloatArgumentHandler.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum FloatArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/FloatArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/InputStreamArgumentHandler.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.io.InputStream;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum InputStreamArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/InputStreamArgumentHandler.java:22]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/IntArgumentHandler.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum IntArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/IntArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/LongArgumentHandler.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum LongArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/LongArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/NullArgumentHandler.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum NullArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/NullArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ObjectArgumentHandler.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum ObjectArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ObjectArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ShortArgumentHandler.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum ShortArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ShortArgumentHandler.java:21]: Is an enum containing one element necessary?

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/StringArgumentHandler.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum StringArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```
Comment (alexandrudanciu) [inline common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/StringArgumentHandler.java:21]: Is an enum containing one element necessary?

**Review by alexandrudanciu — COMMENTED** [review]
This review proposes some improvements related to naming, design patterns. Also some feedback on licensing is provided.

File: common/libraries/clients/postgresql-asset/build.gradle.kts [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    testImplementation("org.testcontainers:postgresql:${testContainersVersion}")
+    testImplementation("org.testcontainers:junit-jupiter:${testContainersVersion}")
+
+    testFixturesImplementation("org.junit.jupiter:junit-jupiter-api:${jupiterVersion}")
+    testFixturesRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:${jupiterVersion}")
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("common-clients-postgresql-assets") {
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/build.gradle.kts:38]: Names are now `postgresql-asset-repository` and `postgresql-asset-index`

File: common/libraries/clients/postgresql/build.gradle.kts [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+    `maven-publish`
+}
+
+val jupiterVersion: String by project
+val postgresqlVersion: String by project
+val apacheCommonsPool2Version: String by project
+val testContainersVersion: String by project
+
+dependencies {
+    implementation("org.postgresql:postgresql:${postgresqlVersion}")
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/build.gradle.kts:27]: Thanks for looking it up :)

File: common/libraries/clients/postgresql/build.gradle.kts [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+    implementation("org.apache.commons:commons-pool2:${apacheCommonsPool2Version}")
+
+    testImplementation("ch.qos.logback:logback-classic:1.2.6")
+    testImplementation("org.testcontainers:postgresql:${testContainersVersion}")
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/build.gradle.kts:31]: Thanks for looking it up :)

File: common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/AssetRepository.java [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.eclipse.dataspaceconnector.clients.postgresql.asset;
+
+import org.eclipse.dataspaceconnector.spi.asset.Criterion;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.jetbrains.annotations.NotNull;
+
+import java.sql.SQLException;
+import java.util.List;
+
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/AssetRepository.java:23]: Good feedback, it now has an java-doc :)

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/LongArgumentHandler.java (hunk 11 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/LongArgumentHandler.java:21]: As this is also discussed in PR #439, where this code was introduced and this PR is based on, I would resolve the comment here. Like this we have the discussion only in one place. I hope this is ok.

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ObjectArgumentHandler.java (hunk 13 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ObjectArgumentHandler.java:21]: As this is also discussed in PR #439, where this code was introduced and this PR is based on, I would resolve the comment here. Like this we have the discussion only in one place. I hope this is ok.

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/StringArgumentHandler.java (hunk 15 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/StringArgumentHandler.java:21]: As this is also discussed in PR #439, where this code was introduced and this PR is based on, I would resolve the comment here. Like this we have the discussion only in one place. I hope this is ok.

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ShortArgumentHandler.java (hunk 14 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ShortArgumentHandler.java:21]: As this is also discussed in PR #439, where this code was introduced and this PR is based on, I would resolve the comment here. Like this we have the discussion only in one place. I hope this is ok.

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/NullArgumentHandler.java (hunk 12 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/NullArgumentHandler.java:21]: As this is also discussed in PR #439, where this code was introduced and this PR is based on, I would resolve the comment here. Like this we have the discussion only in one place. I hope this is ok.

File: common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/ExistsMapper.java [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+package org.eclipse.dataspaceconnector.clients.postgresql.asset.mapper;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ExistsMapper implements RowMapper<Boolean> {
+
+    public static final ExistsMapper INSTANCE = new ExistsMapper();
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/ExistsMapper.java]: Yea, a fast solution that got into the PR somehow. As there is no reason to use a Singleton pattern here, I removed the static field and the class is now created using its constructor.

File: common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/PropertyMapper.java [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+import org.eclipse.dataspaceconnector.clients.postgresql.asset.serializer.EnvelopePacker;
+import org.eclipse.dataspaceconnector.clients.postgresql.asset.types.Property;
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class PropertyMapper implements RowMapper<Property> {
+
+    public static final PropertyMapper INSTANCE = new PropertyMapper();
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/PropertyMapper.java]: Yea, a fast solution that got into the PR somehow. As there is no reason to use a Singleton pattern here, I removed the static field and the class is now created using its constructor.

File: common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/util/Sql.java [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+
+public class Sql {
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/util/Sql.java]: Yea. I renamed it to `PreparedStatementResourceReader` and changed the method prefix to `read-` in all the methods.

File: common/libraries/clients/postgresql-asset/src/test/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/util/TestSql.java [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+
+public class TestSql extends Sql {
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/test/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/util/TestSql.java]: That's true and I removed the inheritance.

File: common/libraries/clients/postgresql/README.md [hunk 24]
```diff
@@ -0,0 +1,43 @@
+# commons-clients-postgresql
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/README.md]: I'd resolve this issues as they are addressed in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/README.md [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
+This library provides two Postgresql clients `SimplePostgresqlClient` and `PooledPostgresqlClient`.
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/README.md]: I'd resolve the PostgreSQL README.md libraries conversations as they are addressed in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/README.md [hunk 26]
```diff
@@ -0,0 +1,43 @@
+# commons-clients-postgresql
+
+This library provides two Postgresql clients `SimplePostgresqlClient` and `PooledPostgresqlClient`. 
+Both implementation provide a unified interface to issue SQL queries towards Postgresql databases.
+Both prevent from SQL injection due to exclusive usage of `java.sql.PreparedStatement`s and yield an interface for transactional SQL queries and automatic transaction rollback while treating exceptions.
+Additionally, the `PooledPostgresqlClient` performs connection pooling, which yields to better performance when the system is under load.
+Connection validation can be configured to identify ill connections.
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/README.md]: I'd resolve the PostgreSQL README.md libraries conversations as they are addressed in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/build.gradle.kts [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+    testImplementation("org.testcontainers:junit-jupiter:${testContainersVersion}")
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("common-clients-postgresql") {
+            artifactId = "common-clients-postgresql"
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/build.gradle.kts:38]: I'd resolve the PostgreSQL build.gradle.library libraries conversations as they are addressed in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/SimplePostgresqlClient.java [hunk 28]
```diff
@@ -0,0 +1,173 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class SimplePostgresqlClient implements PostgresqlClient {
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/SimplePostgresqlClient.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/PostgresqlClient.java [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public interface PostgresqlClient extends AutoCloseable {
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/PostgresqlClient.java:25]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BigDecimalArgumentHandler.java (hunk 2 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BigDecimalArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ByteArgumentHandler.java (hunk 4 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/ByteArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DateArgumentHandler.java (hunk 6 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DateArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DoubleArgumentHandler.java (hunk 7 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/DoubleArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/InputStreamArgumentHandler.java (hunk 9 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/InputStreamArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/IntArgumentHandler.java (hunk 10 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/IntArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BytesArgumentHandler.java (hunk 5 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BytesArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BooleanArgumentHandler.java (hunk 3 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BooleanArgumentHandler.java]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/connection/ConnectionFactoryImpl.java (hunk 1 above)
Comment (DominikPinsel) [reply common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/connection/ConnectionFactoryImpl.java:50]: I'd resolve this conversations as it is already in Denis' PR. Hope this is ok

File: common/libraries/clients/postgresql-asset/src/test/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/RepositoryIntegrationTest.java [hunk 30]
```diff
@@ -0,0 +1,238 @@
...
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Testcontainers
+class RepositoryIntegrationTest {
+
+    @Container
+    @SuppressWarnings("rawtypes")
+    private static final PostgreSQLContainer POSTGRES_CONTAINER = new PostgreSQLContainer("postgres:9.6.12");
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/test/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/RepositoryIntegrationTest.java]: This is the name of a docker image. As it's not a library I'm not sure whether we should put the version into our gradle.properties file. Maybe at a later time, when we have more tests that use docker images, we can create a new Gradle file for this? But I'd not introduce something like this now. I hope this is ok.

File: common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/IdMapper.java [hunk 31]
```diff
@@ -0,0 +1,32 @@
...
+package org.eclipse.dataspaceconnector.clients.postgresql.asset.mapper;
+
+import org.eclipse.dataspaceconnector.clients.postgresql.row.RowMapper;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class IdMapper implements RowMapper<String> {
+
+    public static final IdMapper INSTANCE = new IdMapper();
```
[the remark this answers is not in the document]
Comment (DominikPinsel) [reply common/libraries/clients/postgresql-asset/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/asset/mapper/IdMapper.java]: Yea, a fast solution that got into the PR somehow. As there is no reason to use a Singleton pattern here, I removed the static field and the class is now created using its constructor.

**Pull request closed without merging**